src/compile/data/data.ts

```diff
@@ -115,26 +115,42 @@ export function parseUnitData(model: UnitModel): DataComponent {
  * We only need to add the facet fields as dimensions to the summary component.
  */
 export function parseFacetData(model: FacetModel): DataComponent {
-  // TODO: treat data in the facet and in the child differently
-
   const child = model.child();
-  const dataComponent = child.component.data;
-  delete child.component.data;
 
-  [SOURCE, RAW, STACKED_SCALE].forEach((data) => {
+  // TODO: enforce this in the schema
+  if (child.data() || !empty(child.transform())) {
+    model.addWarning('Facet child should not define data source or transform.');
+  }
+
+  // rename all references to child data, everything should have been merged up
+  [RAW, SOURCE, STACKED_SCALE].forEach((data) => {
     model.renameData(child.dataName(data), model.dataName(data));
   });
 
-  if (dataComponent.aggregate) {
-    aggregate.parseFacet(model, dataComponent.aggregate);
-  }
-  if (dataComponent.stackScale) {
-    stackScale.parseFacet(model, dataComponent.stackScale);
-  }
+  // we should completely move the data component to the facet because
```

Comment:
not sure which part of code this comment try to address  (above? below?  or is it a TODO?)